zcashrpc-typegen/src/tokenize.rs

```diff
@@ -0,0 +1,145 @@
+
+use crate::error;
+use crate::special_cases;
+use error::TypegenResult;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+fn handle_terminal_enum(
+    label: &str,
+    name: &str,
+    called_by_alias: bool,
+) -> TokenStream {
+    let variants = label
+        .strip_prefix("ENUM:")
+        .unwrap()
+        .split(',')
+        .map(|x| x.trim());
+    let variant_idents = variants
+        .clone()
+        .map(|x| {
+            proc_macro2::TokenTree::Ident(crate::callsite_ident(
+                &x.split('-')
+                    .map(crate::capitalize_first_char)
+                    .collect::<String>(),
+            ))
+            .into()
+        })
+        .collect::<Vec<TokenStream>>();
+    let variant_idents_renames = variants
+        .map(|x| format!("#[serde(rename = \"{}\")]", x).parse().unwrap())
+        .collect::<Vec<TokenStream>>();
+    #[rustfmt::skip]
+    let name_tokens = crate::callsite_ident( 
+        &(
+            if called_by_alias {
+                format!("{}Response", name)
+            } else {
+                name.to_string()
+            }
+        )
+    );
+    quote!(
+        #[derive(Debug, serde::Deserialize, serde::Serialize)]
+        pub enum #name_tokens {
+            #(#variant_idents_renames #variant_idents,)*
+        }
+    )
+}
+
+pub(crate) fn value(
```

Comment:
I think this is the only function that needs to be `pub(crate)`, I think the others can all be private. Other than that, this looks great!

Reply:
I will try this.